worker.py

```diff
@@ -1,38 +1,38 @@
 import signal
 import sys
-import socket
+import uuid
+from scheduler_client import SchedulerClient
+from Data.future import Future
+from Data.task import Task
+from logging_provider import logging
 
 class Worker:
-    def __init__(self, scheduler_host, scheduler_port):
+    def __init__(self, scheduler_host, scheduler_port, worker_host, worker_port):
         self.scheduler_host = scheduler_host
         self.scheduler_port = scheduler_port
-        signal.signal(signal.SIGINT, self.sigterm_handler)
+        self.worker_host = worker_host
+        self.worker_port = worker_port
+        self.scheduler_client = SchedulerClient(self.scheduler_host, self.scheduler_port)
+        self.worker_id = "{}:{}".format(worker_host, worker_port)
+        self.scheduler_client.RegisterWorker(self.worker_id)
 
-    def start_client(self):
-        worker_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        worker_socket.connect((self.scheduler_host, self.scheduler_port))
-        print("Worker connected to Scheduler.")
-        while True:
-            task = self.get_task(worker_socket)
-            if task:
-                self.execute_task(task)
-                self.notify_task_completion(task)
+        # Datastore for storing task result in memory. CREATED FOR TESTING FLOW -> NEED TO CHANGE TO FILE BASED SYSTEM LATER (Swarnim's PR)
+        self.dummyFileStore = {}
+        signal.signal(signal.SIGINT, self.sigterm_handler)
 
-    def get_task(self, worker_socket):
-        return worker_socket.recv(1024).decode()
 
-    def execute_task(self, task):
-        print(f"executing task {task}")
-        pass
+    def execute_task(self, task: Task) -> Future:
+        logging.info(f"executing task {task}")
+        resultLocation = str(uuid.uuid4()) #UniqueID
+        self.dummyFileStore[resultLocation] = "TASKDONE".encode()
+        return Future(result_location=resultLocation, host_name=self.worker_host, port=self.worker_port)
```

Comment:
execute task should not return future

Reply:
it should right? Worker should send back the future to the scheduler.

Reply:
I see that you changed this to SubmitTask in latest, this is good now